konrad/core.py

```diff
@@ -177,7 +177,17 @@ def __init__(
         self.logevery = logevery
 
         self.delta = delta
-        self.deltaT = None
+        self.oldT = 0
+        self.oldN = 0
+        self.newT = 0
+        self.newN = 0
+        self.oldDN = 0
+        self.newDN = 0
+        self.oldDDN = 0
+        self.newDDN = 0
+        self.counteq = 0
```

Comment:
I think you can get around all `self.new*` variables. But I will discuss this at a later point in review.

Reply:
Yes, I need to revise the variables. What I am sure is that the DDN are needed: they are second-order differences and they are not obtained from `radiation`.